src/hooks/useDatasetManager.js

```diff
@@ -89,7 +89,9 @@ export const useDatasetManager = () => {
   }
 
   const startProcessingDataset = async (id, options) => {
-    const isCurrentDatasetId = id === datasetId
+    const isCurrentDatasetId = id && id === datasetId
+    const datasetIdToUse =
+      !id || !isCurrentDatasetId ? await createDataset() : id
```

Comment:
```javascript
/* useDatasetManager.js */
// it takes parameters for a dataset ID and download options
  const startProcessingDataset = async (id, options) => { … }
```

This method is also used for one-time experiment downloads. 

Since an one-off has no associated dataset ID by default, `null` was passed as the first argument  `id`.  For this, the `!id ||` check was also added in this PR - commit at f8291c70385c840c72e479a0b4e52cd452a2f33d.

To revert this commit change, I’ve updated the parameter order and made `id`  an optional parameter with its default value (as below), removed the extra logical OR, and adjusted argument order in the files that use this method (6f99fd837d813739aa7ff33f59a048bb2f562ede).

```javascript
/* useDatasetManager.js */
  const startProcessingDataset = async (options, id = null) => { … }
```